Sources/CorePayments/Networking/HTTP.swift

```diff
@@ -24,23 +24,23 @@ class HTTP {
         }
         
         if cachingEnabled, let response = urlCache.cachedResponse(for: urlRequest) {
-            let decodedData = try decoder.decode(T.self, from: response.data)
-            return decodedData
+            if let decodedData = try? decoder.decode(T.self, from: response.data) {
+                return decodedData
+            }
         }
         
         let (data, response) = try await urlSession.performRequest(with: urlRequest)
         guard let response = response as? HTTPURLResponse else {
             throw APIClientError.invalidURLResponseError
         }
         
-        if cachingEnabled {
-            let cachedURLResponse = CachedURLResponse(response: response, data: data)
-            urlCache.storeCachedResponse(cachedURLResponse, for: urlRequest)
-        }
-        
         switch response.statusCode {
         case 200..<300:
             let decodedData = try decoder.decode(T.self, from: data)
+            if cachingEnabled {
+                let cachedURLResponse = CachedURLResponse(response: response, data: data)
+                urlCache.storeCachedResponse(cachedURLResponse, for: urlRequest)
+            }
             return (decodedData)
```

Comment:
Nitpick for organization

```suggestion
            if cachingEnabled {
                let cachedURLResponse = CachedURLResponse(response: response, data: data)
                urlCache.storeCachedResponse(cachedURLResponse, for: urlRequest)
            }
            
            let decodedData = try decoder.decode(T.self, from: data)
            return (decodedData)
```